packages/repository-json-schema/src/filter-json-schema.ts

```diff
@@ -197,20 +197,38 @@ export function getFieldsJsonSchemaFor(
   modelCtor: typeof Model,
   options: FilterSchemaOptions = {},
 ): JsonSchema {
-  const schema: JsonSchema = {
-    ...(options.setTitle !== false && {
-      title: `${modelCtor.modelName}.Fields`,
-    }),
+  const objTitle =
+    options.setTitle !== false
+      ? {
+          title: `${modelCtor.modelName}.Fields`,
+        }
+      : {};
+
+  const properties = Object.keys(modelCtor.definition.properties);
+  const additionalProperties = modelCtor.definition.settings.strict === false;
+  const oneOf: JSONSchema7Definition[] = [];
+
+  oneOf.push({
     type: 'object',
-
-    properties: Object.assign(
+    properties: properties.reduce(
+      (prev, crr) => ({...prev, [crr]: {type: 'boolean'}}),
       {},
-      ...Object.keys(modelCtor.definition.properties).map(k => ({
-        [k]: {type: 'boolean'},
-      })),
     ),
-    additionalProperties: modelCtor.definition.settings.strict === false,
-  };
+    additionalProperties,
+  });
+
+  oneOf.push({
+    type: 'array',
+    items: {
+      type: 'string',
+      enum: properties.length ? properties : undefined,
```

Comment:
Kudos for updating JSON schema too 👍

Originally, I was confused about the `enum` part, but now it makes perfect sense - it allows only real property names in the array.

There is a catch though: models can be defined either as `strict` (no additional properties allowed, typically for SQL) or non-strict (additional properties are allowed, typically for MongoDB). See https://loopback.io/doc/en/lb4/Model.html#supported-entries-of-model-definition For non-strict mode, we need `filter.fields` to allow arbitrary property names, even those that are not defined in the model.

```suggestion
      enum: properties.length && !additionalProperties ? properties : undefined,
```

Please add a test to verify, or update an existing one.

Reply:
oh you're right, maybe instead of use enum i could use example to provide some suggestion about possible values